apps/dashboard/src/components/amount-input.tsx

```diff
@@ -15,6 +15,7 @@ type InputWithSelectProps = {
   className?: string;
   placeholder?: string;
   isReadOnly?: boolean;
+  onValueChange?: (value: string) => void;
```

Comment:
the "amount input" select component for the unit should save the form changes right away and not debounce it